Add buy crypto onboarding banner and refactor banner actions

DRadmir opened the pull request:
Introduces a new 'buyCrypto' banner event and onboarding banner with image assets, updates banner action handling to support multiple action types, and refactors banner view and view models for flexible banner layouts and button actions. Integrates BannerSetupService into onboarding and wallet flows, and updates environment/service injection throughout the app.

Close: https://github.com/gemwalletcom/gem-ios/issues/1126
Close: https://github.com/gemwalletcom/gem-ios/issues/1165
Close: https://github.com/gemwalletcom/gem-ios/issues/1206

TODO:

- [x] Localize
- [x] Test

<img width="320" alt="Simulator Screenshot - iPhone 17 Pro - 2025-09-25 at 16 16 40" src="https://github.com/user-attachments/assets/98d3fe5b-acec-4958-88c3-e477d79302b8" />

DRadmir committed:
Add buy crypto onboarding banner and refactor banner actions

Introduces a new 'buyCrypto' banner event and onboarding banner with image assets, updates banner action handling to support multiple action types, and refactors banner view and view models for flexible banner layouts and button actions. Integrates BannerSetupService into onboarding and wallet flows, and updates environment/service injection throughout the app.

DRadmir committed:
Merge branch 'main' into 1126-add-onboarding-for-buying-crypto

gemcoder21 commented on Features/Onboarding/Sources/ViewsModels/OnboardingViewModel.swift:
```diff
@@ -4,19 +4,26 @@ import Foundation
 public final class OnboardingViewModel {
     let walletService: WalletService
     let nameService: any NameServiceable
+    let bannerSetupService: BannerSetupService
```
`bannerSetupService` should not be part of onboard flow. You need to setup callback from create / import wallet flow to let you know that import or create is done, here is the result, otherwise this is too much responsibility for onboarding to handle

gemcoder21 commented on Packages/PrimitivesComponents/Sources/Components/BannerView.swift:
```diff
@@ -39,30 +33,52 @@ public struct BannerView: View {
+            }
+            .scrollTargetLayout()
+        }
+        .scrollTargetBehavior(.paging)
```
we don't need any scrolling, keep it simple

gemcoder21 commented on Packages/Style/Sources/Images.swift:
```diff
@@ -196,6 +196,10 @@ public enum Images {
     }
+    
+    public enum Banners {
+        public static let pengu = Image(.gmPengu)
```
```suggestion
        public static let pengu = Image(.gmPengu)
```

there is should not be any pengu, clean and simple icon to represent what you want to do

gemcoder21 commented on Packages/SystemServices/BannerService/BannerSetupService.swift:
```diff
@@ -46,4 +46,10 @@ public struct BannerSetupService: Sendable {
+    
+    public func setupBuyCryptoBanner(walletId: WalletId) throws {
+        try store.addBanners([
+            NewBanner.buyCrypto(walletId: walletId)
```
```suggestion
            NewBanner.buyCrypto(walletId: walletId)
```

it's not related to buy crypto, this is onboarding flow and not domain driven functionality

DRadmir committed:
Merge branch 'main' of github.com:gemwalletcom/gem-ios into 1126-add-onboarding-for-buying-crypto

DRadmir committed:
Refactor banner handling and remove unused assets

Updated banner display logic to show only the priority banner in asset and wallet scenes. Refactored related view models and removed unused 'gm-pengu' banner assets and references. Updated onboarding banner setup and standardized banner image usage.

DRadmir committed:
Add priority banner tests and BannerState sorting tests

Added tests for priorityBanner logic in AssetBannersViewModelTests and WalletSceneViewModelTests. Removed redundant allBanners test from AssetSceneViewModelTests. Introduced BannerState sorting and priority tests in Primitives. Updated Onboarding package dependencies to include BannerService.

DRadmir committed:
Refactor onboarding banner and banner sorting logic

Replaces the 'buyCrypto' banner event with a new 'onboarding' event throughout the codebase. Introduces Comparable conformance for Banner and BannerEvent to enable consistent sorting. Refactors WalletScene banner logic into a dedicated WalletSceneBannersViewModel, updates related tests, and removes unused BannerSetupService dependencies from onboarding flows. Improves BannerView layout and button styling.

DRadmir committed:
Update core

DRadmir committed:
Merge branch 'main' of github.com:gemwalletcom/gem-ios into 1126-add-onboarding-for-buying-crypto

# Conflicts:
#	core

DRadmir commented:
@gemcoder21 Please review sort priority Packages/Primitives/Sources/Extensions/BannerEvent+Primitives.swift

DRadmir committed:
Adjust close icon padding in BannerView

Changed the padding of the close (xmark) icon from .small to .tiny in BannerView for a more compact appearance.

DRadmir committed:
Merge branch 'main' into 1126-add-onboarding-for-buying-crypto

DRadmir committed:
Merge branch 'main' of github.com:gemwalletcom/gem-ios into 1126-add-onboarding-for-buying-crypto

# Conflicts:
#	core

DRadmir committed:
Update core submodule to latest commit

Advanced the core submodule reference from 234597e1 to 0f84f989 to incorporate the latest changes from the core repository.

DRadmir committed:
Add onboarding banner localization

Introduces localized strings for onboarding banner title and description across all supported languages. Adds new perpetual trading actions: modify, reduce position, and increase position. Updates BannerViewModel to use localized onboarding strings. Minor code formatting fix in Banner.swift.

DRadmir committed:
Merge branch 'main' of github.com:gemwalletcom/gem-ios into 1126-add-onboarding-for-buying-crypto

# Conflicts:
#	Packages/Localization/Sources/Resources/ar.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/bn.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/cs.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/da.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/de.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/en.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/es.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/fa.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/fil.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/fr.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/he.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/hi.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/id.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/it.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/ja.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/ko.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/ms.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/nl.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/pl.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/pt-BR.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/ro.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/ru.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/sw.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/th.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/tr.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/uk.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/ur.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/vi.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/zh-Hans.lproj/Localizable.strings
#	Packages/Localization/Sources/Resources/zh-Hant.lproj/Localizable.strings
#	Packages/Preferences/Sources/WalletPreferences.swift
